### PR TITLE
Fix/update self registration patient email validation

### PR DESCRIPTION
**Pull request by Zhaolian-Zhou**
Add back some validation to ensure error messages don't missing when more than one mandatory fields are empty.

Also, I mixed the country dropdownlist change here as well.

### CODE REVIEW

**Review by apigram — COMMENTED** [review]
Would this cause the issue where valid fields would appear invalid?

**Comment by Zhaolian-Zhou** [comment]
As far as I tested, no such issue appear again, such as the DOB error when email is left blank.
You can do more tests in case I missed some situations. @apigram @marshl @Vegemash

**Review by apigram — APPROVED** [review]

File: protected/controllers/PatientController.php [hunk 1]
```diff
@@ -1668,11 +1668,19 @@ private function performPatientSave(Contact $contact, Patient $patient, Address
...
                 } else {
+                    $address->validate();
                     // patient or address failed to save
                     $transaction->rollback();
 
                 }
             } else {
+                //to show validation error messages to the user
+                $address->validate();
+                $patient->validate();
```
Comment (marshl) [inline protected/controllers/PatientController.php]: We might have to test this a bit more. $patient->validate() was what caused the original issue

File: protected/views/patient/_form_address.php [hunk 2]
```diff
@@ -60,8 +60,11 @@
 <div class="row field-row">
     <div class="large-3 column"><?php echo $form->labelEx($address,'country_id'); ?></div>
     <div class="large-4 column end">
-
-        <?php echo $form->dropDownList($address,'country_id', $countries, array('options'=>array(array_search('Australia', $countries)=>array('selected'=>true)))); ?>
+        <?php if (isset($address["country_id"])) {
```
Comment (marshl) [inline protected/views/patient/_form_address.php]: This code is from another commit/pull-request. You might have to be rebase this commit from the base branch locally, and force push back to this branch before you'll be able to rebase and merge

Comment (Zhaolian-Zhou) [reply protected/views/patient/_form_address.php]: Seems I messed up rebase things, I will create a new pull request.
I hate rebase :(

**Pull request closed without merging**